python/kvikio/tests/test_aws_s3.py

```diff
@@ -0,0 +1,137 @@
+# Copyright (c) 2024, NVIDIA CORPORATION. All rights reserved.
+# See file LICENSE for terms.
+
+import multiprocessing as mp
+import os
+import socket
+import time
+from contextlib import contextmanager
+
+import pytest
+
+import kvikio
+
+# TODO: remove before PR merge. Trigger CI error if the remote module wasn't built
+import kvikio._lib.remote_handle  # isort: skip
+
+moto = pytest.importorskip("moto", minversion="3.1.6")
+boto3 = pytest.importorskip("boto3")
```

Comment:
I don't believe these should be `importorskip`. I understand the intention of not running the tests if S3 support is not available, but if it is available, these should be required.

Reply:
I like to use `pytest.importorskip` to allow users run pytest without optional packages installed. 
However, we could introduce (in another PR) a dedicated test that tries to include all optional packages and fails if any of them isn't available.

Reply:
Sure. My point is that in this particular case, if S3 support is present (as determined by `is_remote_file_available()` above), then `moto` and `boto3` should be required to test it. The test itself is optional, but if we're going to run it, then `moto` and `boto3` are required. (Note that they've both been added as optional dependencies in the `test` category already.)

Reply:
I am not sure, what if a user build/install kvikio with s3 support but doesn’t have `moto` or `boto3`? In this case, I just want pytest to skip the tests. The same goes for installations without `cupy`, `numpy`, or `zarr`.

Reply:
Good point on that scenario. Not how I'd do it, but I suppose it's reasonable. I'll leave it to you to decide.

Reply:
Using `importorskip` like that makes our test suites prone to accidentally skipping tests. I tend to lean the same way as Kyle here: it is reasonable for the test suites to have dependencies, and we require those for the whole test suite to pass. Users missing a test dependency can skip the affected tests manually or install the needed packages if they’re looking for a green/passing result.

Reply:
OK, I see your points, let's make them `import`